packages/engine/src/datastore/batch/context.rs

```diff
@@ -26,6 +26,12 @@ const UPPER_BOUND_DATA_SIZE_MULTIPLIER: usize = 3;
 pub type AgentIndex = (u32, u32);
 pub type MessageIndex = (u32, u32, u32);
 
+/// Contains the information required to build the `context` object accessible in the language
+/// runners.
+///
+/// The `ContextBatch` can refer to the **Agent** and **Message Pool** within the **Context**
```

Comment:
I don't think this is true? 

@litvand the `previous_state: StatePools` field on the `Context` struct are the previous step's agent state and `MessagePool` right?

The `ContextBatch` itself is computed by the context packages. There's only one per simulation run.

We could perhaps give examples like "api_request" or "neighbors" but I would prefer we just link to `crate::simulation::package::context::Package`, and refer to the `get_empty_arrow_columns` and `run` methods.

Reply:
I think this is true. There is one context batch per simulation run, which can refer to the previous step's state agent and message pool, which are in the `previous_state` field.

Reply:
Right but the `Batch` struct (exported as `ContextBatch`) these lines are documenting doesn't have the previous steps agent state and message pool does it? I'm saying these lines are wrongly documenting this specific struct, not the context as a whole.

Those two things are stored in the `previous_state` field on a _different_ struct no?

Perhaps I'm missing something

Reply:
I see your point. I guess it doesn't refer to them in the sense that it doesn't have references to them, but the context batch contains indices into the pools, e.g. the group index and index within the group of some neighbor within the previous step's agent pool.

Reply:
Ahah I just understood the wording. I thought it meant the term "ContextBatch" can refer to those things, rather than the contents of the `ContextBatch` refers to data within the Agent and Message pools.

My mistake!

Reply:
```suggestion
/// Data within the `ContextBatch` can refer to the **Agent** and **Message Pools** within the
/// **Context**
```